workloads/upgrade-perf/rosa_upgrade.sh

```diff
@@ -0,0 +1,139 @@
+export rosa_version_channel=${ROSA_VERSION_CHANNEL:=stable}
+_es_index=${ES_INDEX:=managedservices-install-timings}
+
+rosa_upgrade(){
+  if [ ${rosa_version_channel} == "nightly" ] ; then
+    echo "ERROR: Invalid channel group. Nightly versions cannot be upgraded. Exiting..."
+    exit 1
+  fi
+  echo "ROSA Cluster: ${ROSA_CLUSTER_NAME}"
+  echo "ROSA Environment: ${rosa_environment}"
+  echo "ROSA Channel Group: ${rosa_version_channel}"
+  export HOME=${HOME:=$PWD}
+  rosa login --env=${rosa_environment} --token=${ROSA_TOKEN}
+  if [ $(rosa list clusters | awk '{print $2}' | sed -e 1d | grep ${ROSA_CLUSTER_NAME}) != ${ROSA_CLUSTER_NAME} ] ; then
+    echo "ERROR: Cluster ${ROSA_CLUSTER_NAME} not found on rosa list clusters results. Exiting..."
+    exit 1
+  fi
+  if [ ! -z ${LATEST} ] && [ -z ${TOVERSION} ] ; then
+    echo "INFO: Getting latest version available to upgrade..."
+    VERSION=$(rosa list upgrade -c ${ROSA_CLUSTER_NAME} | sed -e 1d | head -1 | awk '{print $1}')
+  elif [ ! -z ${TOVERSION} ] && [ -z ${LATEST} ] ; then
+    echo "INFO: Checking if ${TOVERSION} version is available to upgrade..."
+    VERSION=$(rosa list upgrade -c ${ROSA_CLUSTER_NAME} | sed -e 1d | awk '{print $1}' | grep ${TOVERSION})
+  elif [ -z ${TOVERSION} ] && [ -z ${LATEST} ] ; then
+    echo "ERROR: Only LATEST or TOVERSION can be defined. Exiting..."
+    exit 1
+  else
+    echo "ERROR: Missing upgrade strategy. No LATEST or TOVERSION are defined. Exiting..."
+    exit 1
+  fi
+  if [ -z ${VERSION} ] ; then
+    echo "ERROR: No version to upgrade found running rosa list upgrade -c ${ROSA_CLUSTER_NAME}"
+    exit 1
+  else
+    echo "INFO: Upgrading cluster to ${VERSION} version..."
+  fi
+  CURRENT_VERSION=$(oc get clusterversion | grep ^version | awk '{print $2}')
+  rosa upgrade cluster -c ${ROSA_CLUSTER_NAME} --version=${VERSION} -y --schedule-date $(date +%Y-%m-%d) --schedule-time $(date -u --date="+6 minutes" +%H:%M)
+  # Sleep 6 minutes, rosa upgrade dont let to schedule an upgrade in less than 5 minutes from now
+  sleep 360
+  oc delete pods -n openshift-insights --all
+  oc delete pods -n openshift-managed-upgrade-operator --all
+  rosa_control_plane_upgrade_active_waiting ${VERSION}
+  if [ $? -eq 0 ] ; then
+    CONTROLPLANE_UPGRADE_RESULT="OK"
+  else
+    CONTROLPLANE_UPGRADE_RESULT="Failed"
+  fi
+  rosa_workers_active_waiting
+  if [ $? -eq 0 ] ; then
+    WORKERS_UPGRADE_RESULT="OK"
+  else
+    WORKERS_UPGRADE_RESULT="Failed"
+  fi
+  rosa_upgrade_index_results ${CONTROLPLANE_UPGRADE_DURATION} ${CONTROLPLANE_UPGRADE_RESULT} ${WORKERS_UPGRADE_DURATION} ${WORKERS_UPGRADE_RESULT} ${CURRENT_VERSION} ${VERSION}
+  exit 0
+}
+
+rosa_workers_active_waiting(){
+  start_time=$(date +%s)
+  WORKERS=$(oc get node --no-headers -l node-role.kubernetes.io/workload!="",node-role.kubernetes.io/infra!="",node-role.kubernetes.io/worker="" 2>/dev/null | wc -l)
+  # Giving 5 minutes per worker
```

Comment:
Can we variabilize the 5 min per worker?

Reply:
ok done